NektarDriftwave/README.md

```diff
@@ -0,0 +1,175 @@
+
+### Simulating from model
+
+To generate a simulated observation sequence from the model we can use the [`simulated_observations_from_model` function](https://team-raddish.github.io/ParticleDA.jl/stable/#ParticleDA.simulate_observations_from_model) exported by `ParticleDA`. We need to initialise an instance of the model using the `init` method specified in the `NektarDriftwave` module, passing in a dictionary of any model parameters we wish to adjust from their default values - in most cases this will include setting the parameters `nektar_bin_directory` and `driftwave_solver_bin_directory` to point to the system specific directories in which the built Nektar++ and `nektar-driftwave` binaries can be found. The code example below simulates the model for 100 (equispaced) observation times, writing the resulting simulated observation sequence to a HDF5 file `observations.h5` in the current working directory.
+
+```Julia
+using ParticleDA
+using HDF5
+
+include("src/nektar_driftwave.jl")
+
+num_observation_times = 100
+
+model_parameters_dict = Dict(
+    "nektar_bin_directory" => "/path/to/nektar/build/dist/bin",
+    "driftwave_solver_bin_directory" => "/path/to/nektar-driftwave/build/dist",
+)
+
+model = NektarDriftwave.init(model_parameters_dict)
+
+observation_sequence = h5open("observations.h5", "w") do output_file
+    simulate_observations_from_model(
+        model, num_observation_times; output_file
+    )
+end
+```
+
+### Distributed filtering using MPI
+
+To perform filtering, that is estimation of the distribution of the model state at each observed time given the observations up to that time, we can use the [`run_particle_filter` function](https://team-raddish.github.io/ParticleDA.jl/stable/#ParticleDA.run_particle_filter) exported by ParticleDA. ParticleDA supports distributing the filtering run using a _message passing interface_ (MPI) implementation, with each process sampling new values for one or more state particles in the ensemble from the proposal distributions in parallel, with MPI communication operations used to perform the resampling step which involves synchronization across the processes. The code example below would perform filtering for the NektarDriftwave state space model for a sequence of observations recorded in a HDF5 file `observations.h5`, and write the filtering results to a HDF5 file `filter_results.h5`. It is assumed the code is launched using [the `mpiexecjl` Julia wrapper](https://juliaparallel.org/MPI.jl/latest/usage/#Julia-wrapper-for-mpiexec)  to run over multiple MPI ranks (processes) with each rank hosting a fixed number of particles (by default here one particle per rank). We specify to use the bootstrap filter type and to compute both mean and variance summary statistics from the ensemble at each observation time.
+
+```Julia
+using ParticleDA
+using HDF5
+using MPI
+
+include("src/nektar_driftwave.jl")
```

Comment:
If refactored as a package, change to `using NektarDriftwave` - would also remove need to be in correct directory when running this and other examples.